.gitignore

```diff
@@ -89,3 +89,6 @@ ENV/
 /.pydevproject
 /.settings
 .vscode
+
+# GIT stuff, needs for docker
+/.git
```

Comment:
Seems odd—wouldn't this be more logical to include in `.dockerignore`, not `.gitignore`?

Reply:
Nope, because `.dockerignore` should be inside git (and it is a symlink to `.gitignoe`) ;)

Why I added it? Because this is an easy way to add `.git` inside `.dockerignore`

Reply:
I think @sampsyo is right here still - `.git` belongs in the `.dockerignore` really ([it seems pretty common to do this](https://github.com/search?l=Ignore+List&q=.git+filename%3A.dockerignore&type=Code)).

Reply:
@jackwilsdon it requires to create this `.dockerfile`; right now I just make a symlink from `.gitingore` to `.dockeringore` because otherwise it requires to add everything into two places

Reply:
Ah I didn't realise that `.dockerignore` was symlinked to `.gitignore` - I'm a bit hesitant about linking them like this as it doesn't really seem to be a common practice.

Reply:
@jackwilsdon but they very similar by idea :)

Reply:
the dockerignore and gitignore should be separated because it makes sense (to me) to have CI/CD config or the .github folder ignored by docker but not by git, for example.

Also, for "production-ready" Dockerfiles, I usually use an allowlist, so they look a bit different than my .gitignore, like
```ignore
*
!source_code
!dependencies
```

Reply:
@GuilhermeHideki and your way will include to doker image all `.pyc` files if you make `pip install -e .` ;)

Reply:
@catap This should help
```ignore
# Exclude everything
*
# setup.py dependency
!README.rst
# Make sure that source folters are in dockerfile
!beets
!beetsplug
# Ignore cache files
**/__pycache__
**/*.pyc
# Include code / source files
!**/*.py
!**/*.yaml
!**/*.sh
```

Reply:
@GuilhermeHideki the idea was to not duplicated ignored/not ignored files.

The goal was to have everything that is ignored in one place.